### PR TITLE
feat(extension): add extension to quick search and copy common regula…

### PR DESCRIPTION
**Pull request by musienkoyuriy**
# Search RegExp

## Description

This extension allows you to quickly find the most common JS regular expressions including payments validation, password strenght, dates, time, ZIP codes etc.

## Screencast

search-regexp-zip-codes

search-regexp-show-in-browser

search-regexp-default

search-regexp-filtered

## Supported Commands

1. Search regular expressions
2. Copy to clipboard
3. Open in browser

## How to use

1. Search regular expressions by name or category.
2. Filter items by category using dropdown near the search field (Cmd+P).
3. If you need to find a regular expression for certain ZIP code, find `Zip Codes` in list and go to zip codes list.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/search-regexp/src/components/CategoriesDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { List } from "@raycast/api";
+import { Category } from "../types";
+
+export default function CategoriesDropdown({
+  categories,
+  onCategoryChange,
+}: {
+  categories: Category[];
+  onCategoryChange: (newCategory: string) => void;
```
Comment (Thegrep01) [inline extensions/search-regexp/src/components/CategoriesDropdown.tsx:9]: It's always better to create exportable props interface like 
`export interface CategoriesDropdownProps { ... }`

File: extensions/search-regexp/src/components/ZipCodeList.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+enum Loading {
+  IDLE,
+  LOADING,
+  LOADED,
+}
+
+export default function ZipCodesList({ expressions }: { expressions: MappedExpression[] }): JSX.Element {
+  const [loading, setLoading] = useState<Loading>(Loading.IDLE);
+  const [zipCodes, setZipcodes] = useState<MappedExpression[] | null>(null);
+  const [filteredZipCodes, setFilteredZipCodes] = useState<MappedExpression[]>([]);
```
Comment (Thegrep01) [inline extensions/search-regexp/src/components/ZipCodeList.tsx:24]: when you have a lot of states in component, you can use [useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer)

File: extensions/search-regexp/src/components/ZipCodeList.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      }, {});
+
+      setZipcodes(
+        expressions.map((expression) => ({
+          ...expression,
+          description: mappedResponse[expression.name]?.country || "",
+        })) as unknown as MappedExpression[]
+      );
+    })();
+  }, []);
```
Comment (Thegrep01) [inline extensions/search-regexp/src/components/ZipCodeList.tsx:76]: the complex logic parts you can divide into separate custom hooks ex. useZipcodes() where you can put and then reuse or just keep components clear

**Comment by pernielsentikaer** [comment]
Hi @musienkoyuriy 👋 

Do you mind checking my corrections, are they okay?

**Comment by musienkoyuriy** [comment]
hi @pernielsentikaer! Yep, I checked your changes, it's okay, thank you!

**Comment by musienkoyuriy** [comment]
@pernielsentikaer since now we display regexps instead of duplicated categories in items subtitle, I've pushed commit for truncating them because a lot of regexps are too long

**Pull request merged**